consensus/polybft/proposer_calculator_test.go

```diff
@@ -0,0 +1,510 @@
+package polybft
+
+import (
+	"bytes"
+	"math"
+	"testing"
+
+	bls "github.com/0xPolygon/polygon-edge/consensus/polybft/signer"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/hashicorp/go-hclog"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestProposerCalculator_SetIndex(t *testing.T) {
+	t.Parallel()
+
+	validators := newTestValidatorsWithAliases([]string{"A", "B", "C", "D", "E"}, []uint64{10, 100, 1, 50, 30})
+	metadata := validators.getPublicIdentities()
+
+	vs, err := validators.toValidatorSet()
+	require.NoError(t, err)
+
+	snapshot := NewProposerSnapshot(1, metadata)
+	pc := NewProposerCalculatorFromSnapshot(snapshot, hclog.NewNullLogger())
+
+	// validate no changes to validator set positions
+	for i, v := range vs.Accounts() {
+		assert.Equal(t, metadata[i].Address, v.Address)
+	}
+
+	proposer, err := pc.CalcProposer(0, 1)
+	require.NoError(t, err)
+	assert.Equal(t, proposer, metadata[1].Address)
+	// validate no changes to validator set positions
+	for i, v := range vs.Accounts() {
+		assert.Equal(t, metadata[i].Address, v.Address)
+	}
+}
+
+func TestProposerCalculator_CalculateProposer(t *testing.T) {
+	t.Parallel()
+
+	validators := newTestValidatorsWithAliases([]string{"A", "B", "C", "D", "E"}, []uint64{1, 2, 3, 4, 5})
+	metadata := validators.getPublicIdentities()
+
+	vs, err := validators.toValidatorSet()
+	require.NoError(t, err)
+
+	snapshot := NewProposerSnapshot(0, metadata)
+	pc := NewProposerCalculatorFromSnapshot(snapshot, hclog.NewNullLogger())
+
+	assert.Equal(t, int64(15), vs.totalVotingPower)
+
+	currProposerAddress, err := pc.CalcProposer(0, 0)
```

Comment:
Could you do a table test here?

Reply:
https://github.com/0xPolygon/polygon-edge/pull/1018#discussion_r1051650359